junit-runtime/src/main/scala/org/junit/internal/ComparisonCriteria.scala

```diff
@@ -0,0 +1,117 @@
+/*
+ * Ported from https://github.com/junit-team/junit
+ */
+package org.junit.internal
+
+import org.junit.Assert
+
+abstract class ComparisonCriteria {
+
+  def arrayEquals(message: String, expecteds: AnyRef, actuals: AnyRef): Unit =
+    arrayEquals(message, expecteds, actuals, true)
+
+  private object Arrays {
+    def deepEquals(a1: Array[AnyRef], a2: Array[AnyRef]): Boolean = {
+      if ((a1 eq a2) || a1.sameElements(a2))
+        true
+      else if (a1 == null || a2 == null || a1.length != a1.length)
+        false
+      else {
+        (a1 zip a2).forall {
+          case (e1, e2) if (e1 eq e2) || e1 == e2 => true
+          case (e1: Array[AnyRef], e2: Array[AnyRef]) => deepEquals(e1, e2)
```

Comment:
Actually I wrote this before implementing `Arrays`, now I can use `Arrays.deepEquals`.